src/screens/WebsocketResearchScreen/SampleChatScreen.tsx

```diff
@@ -4,75 +4,87 @@
 import BaseChatItem from '../../components/AnonymousChat/BaseChatItem';
-import useAnonymousChatScreenHook from '../../hooks/screen/useAnonymousChatScreenHook';
+import ChatDetailHeader from '../../components/AnonymousChat/ChatDetailHeader';
+import useChatScreenHook from '../../hooks/screen/useChatScreenHook';
+import {ANONYMOUS} from '../../hooks/core/constant';
 import {colors} from '../../utils/colors';
 
 const {height} = Dimensions.get('window');
 
+export const styles = StyleSheet.create({
+  keyboardAvoidingView: {
+    flex: 1,
+    backgroundColor: colors.white
+  },
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: height - 85
+  },
+  chatContainer: {
+    display: 'flex',
+    height: '100%',
+    marginBottom: 72
+  },
+  inputContainer: {
+    backgroundColor: colors.white,
+    position: 'absolute',
+    bottom: 0,
+    // height: 50,
+    left: 0,
+    right: 0,
+    zIndex: 100,
+    padding: 8,
+    paddingBottom: 16,
+    borderTopColor: colors.lightgrey,
+    borderTopWidth: 1
+  }
+});
+
 const SampleChatScreen = () => {
+  const flatlistRef = React.useRef<FlatList>();
   const {selectedChannel, chats, goBackFromChatScreen, goToChatInfoScreen, sendChat} =
-    useAnonymousChatScreenHook();
-  const styles = StyleSheet.create({
-    keyboardAvoidingView: {
-      flex: 1,
-      backgroundColor: colors.white
-    },
-    container: {
-      display: 'flex',
-      flexDirection: 'column',
-      height: height - 85
-    },
-    chatContainer: {
-      display: 'flex',
-      height: '100%',
-      marginBottom: 72
-    },
-    inputContainer: {
-      backgroundColor: colors.white,
-      position: 'absolute',
-      bottom: 0,
-      // height: 50,
-      left: 0,
-      right: 0,
-      zIndex: 100,
-      padding: 8,
-      paddingBottom: 16,
-      borderTopColor: colors.lightgrey,
-      borderTopWidth: 1
-    }
-  });
+    useChatScreenHook(ANONYMOUS);
 
   const renderChatItem = React.useCallback(({item, index}) => {
     return <BaseChatItem item={item} index={index} />;
   }, []);
 
+  const scrollToEnd = () => {
+    flatlistRef.current?.scrollToEnd();
+  };
+
   return (
     <KeyboardAvoidingView
       behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
       style={styles.keyboardAvoidingView}
       keyboardVerticalOffset={-500}>
-      <AnonymousChatHeader
-        onAvatarPress={goToChatInfoScreen}
-        onBackPress={goBackFromChatScreen}
-        onThreeDotPress={goToChatInfoScreen}
-        avatar={selectedChannel?.channelPicture}
-        user={selectedChannel?.name}
-      />
+      {selectedChannel ? (
+        <ChatDetailHeader
+          onAvatarPress={goToChatInfoScreen}
+          onBackPress={goBackFromChatScreen}
+          onThreeDotPress={goToChatInfoScreen}
+          avatar={selectedChannel?.channelPicture}
+          user={selectedChannel?.name}
+          type={ANONYMOUS}
+        />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ChatDetailHeader` component is now being used instead of the `AnonymousChatHeader` component. This is a good change as it allows for more detailed information to be displayed in the header.

```diff
- <AnonymousChatHeader
+ <ChatDetailHeader
  onAvatarPress={goToChatInfoScreen}
  onBackPress={goBackFromChatScreen}
  onThreeDotPress={goToChatInfoScreen}
  avatar={selectedChannel?.channelPicture}
  user={selectedChannel?.name}
+ type={ANONYMOUS}
/>
```

87:
The `AnonymousInputMessage` component now takes an additional `type` prop. This is a good change as it allows the component to adapt its behavior based on the type of chat.

```diff
- <AnonymousInputMessage onSendButtonClicked={sendChat} />
+ <AnonymousInputMessage onSendButtonClicked={sendChat} type={ANONYMOUS} />
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->